apps/backend/app/api/routes/transformations.py

```
@@ -1511,4 +1522,372 @@ async def clear_history(
+        )
+
+
+@router.post(
+    "/datasets/{dataset_id}/bulk-apply",
+    response_model=BulkTransformationJobResponse
+)
+async def apply_bulk_transformation(
+    dataset_id: str,
+    request: BulkTransformationRequest,
+    current_user_id: str = Depends(get_current_user_id),
+    bulk_service: "BulkTransformationService" = Depends(get_bulk_transformation_service)
```

Ruff (0.14.10)

Undefined name `BulkTransformationService`

(F821)